src/cli.py

```diff
@@ -0,0 +1,45 @@
+import sys
+import click
+from pathlib import Path
+from src.comparsion import compare_and_output
+
+def validate_paths(input_path: str, output_path: str) -> None:
+    input_path = Path(input_path)
+    
+    if not input_path.is_dir():
+        raise FileNotFoundError(f"Input path does not exist: {input_path}")
+    
+    if output_path:
+        output_path = Path(output_path)
+        if not output_path.is_dir():        
+            output_path.mkdir(parents = True, exist_ok = True)
+            print(f"Attention: new folder created at: {output_path}")
+
+    if not output_path:
+        output_path = Path(Path.cwd(), "output")
+
+        if not output_path.is_dir():        
+            output_path.mkdir(parents = True, exist_ok = True)
+            print("Default output folder created at:", output_path)
+
+    return input_path, output_path
+
+@click.command()
+@click.option('--input', type = str, required = True, help = 'Directory containing three folders of source data.')
+@click.option('--output', type = str, help = 'Directory to store output files. The input is optional.')
+@click.option('--codeql', type = str, required = True, help = 'Relative path to truncate from file names in the CodeQL data.')
+@click.option('--clojure', type = str, help = 'Relative path to truncate from file names in the Clojure data.')
+@click.option('--coccinelle', type = str, required = True, help = 'Relative path to truncate from file names in the Coccinelle data.')
+@click.option('--whichtwo', type = str, required = True, help = "This option selects two, or all, of the three tools to be compared. Four possible inputs are supported: 'CLJvCQL', 'CLJvCNL', 'CQLnCNL', and 'all'. Anything else will behave as the 'all' option.")
```

Comment:
I think that it makes sense, as this would allow the user to only compare two inputs, even if all 3 files exist. You can use click.Choice (https://click.palletsprojects.com/en/8.1.x/options/#choice-options). Additionally, we still need to automatically check which input files exist. So, if this is not provided, but only coccinelle and clojure outputs exist, se to CLJvCN.